lib/modules/dosomething/dosomething_user/dosomething_user_social_auth.inc

```diff
@@ -0,0 +1,98 @@
+<?php
+
+function dosomething_user_make_facebook_user($account_details) {
+  $edit = [
+    'mail' => $account_details['email'],
+    'name' => $account_details['email'],
+    'pass' => user_password(),
+    'status' => 1,
+    'created' => REQUEST_TIME,
+  ];
+
+  dosomething_user_set_fields($edit, $account_details);
+
+  $user = user_save(null, $edit);
```

Comment:
in this case $user is an anonymous user, so in order to make it work in this function,

> If you want to create a new user account, you can set $account->is_new to TRUE or omit the $account->uid field.
https://api.drupal.org/api/drupal/modules%21user%21user.module/function/user_save/7.x

I saw we used something similar to this in the member resource API https://github.com/DoSomething/phoenix/blob/dev/lib/modules/dosomething/dosomething_api/resources/member_resource.inc#L218

but I figured NULL made more sense then `''`